physionet-django/project/views.py

```diff
@@ -2285,21 +2286,12 @@ def generate_signed_url(request, project_slug):
     filename = request.POST.get('filename')
     size = request.POST.get('size')
 
-    if not filename or not size:
-        return JsonResponse({'detail': 'You must provide the filename and the size.'}, status=400)
-
-    if not size.isnumeric():
-        return JsonResponse({'detail': 'The file size must be a numeric value.'}, status=400)
-
-    size = int(size)
-    if size <= 0:
-        return JsonResponse({'detail': 'The file size cannot be a negative value.'}, status=400)
-
     try:
-        validate_filename(filename)
+        validate_gcs_bucket_object(filename, size)
     except ValidationError as e:
         return JsonResponse({'detail': e.messages}, status=400)
 
+    size = int(size)
```

Comment:
Why is this cast necessary?

Reply:
I think I get it. This makes things a bit awkward but seems hard to avoid. I actually did not notice that we can create a single function both for size and filename validation but it does seem cleaner this way.

Reply:
Oh yes, because there is 

1. `if size > project.get_storage_info().remaining:` which checks if the project has space remaining and expects and integer
2. `generate_signed_url_helper` also expects an integer.

So i thought typecasting once would be better than typecasting twice.

or another option was to bring the size validation on the view and it will do typecasting before validation.

Reply:
oh yes, i took your suggestion one step further. 

its weird to avoid type casting unless we want to bring the size validation inside the view, or do typecasting directly on the code when needed

Reply:
It's weird it's not sent as a number 🤷 Your solution makes sense though.

Reply:
Let me know if you think separating validation function for size and name makes more sense

Reply:
> number

Ah yes, we also have a validation that does `size.isnumeric()` , so it looks like size is always received  as string.